src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2693 @@
+        avgBpsIn <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsIn
+        avgBpsOut <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsOut
+        return NetworkInfo{..}
+
+data NetworkInfo = NetworkInfo
+    { nodeId :: !Text
+    , peerTotalSent :: !Integer
+    , peerTotalReceived :: !Integer
+    , avgBpsIn :: !Integer
+    , avgBpsOut :: !Integer
+    }
+
+data NodeInfo = NodeInfo
+    { peerVersion :: !Text
+    , localTime :: !Timestamp
+    , peerUptime :: !Duration
+    , networkInfo :: !NetworkInfo
+    , details :: !(Maybe NodeDetails)
+    }
+
+instance FromProto Proto.ChainParametersV0 where
+    type Output' Proto.ChainParametersV0 = Parameters.ChainParameters' 'ChainParametersV0
+    fromProto cParams = do
+        _cpElectionDifficulty <- fromProto =<< cParams ^? ProtoFields.electionDifficulty
+        _cpExchangeRates <- do
+            euroPerEnergy <- fromProto =<< cParams ^? ProtoFields.euroPerEnergy
+            microCcdPerEuro <- fromProto =<< cParams ^? ProtoFields.microCcdPerEuro
+            return $ Parameters.makeExchangeRates euroPerEnergy microCcdPerEuro
+        _cpCooldownParameters <-
+            fmap Parameters.CooldownParametersV0
+                . fromProto
+                =<< cParams ^? ProtoFields.bakerCooldownEpochs
+        -- FIXME: I don't see this (required field) anywhere in the protobuf message.
+        _cpTimeParameters <- Nothing
+        _cpAccountCreationLimit <- fromProto =<< cParams ^? ProtoFields.accountCreationLimit
+        _cpRewardParameters <- do
+            _rpMintDistribution <- fromProto =<< cParams ^? ProtoFields.mintDistribution
+            -- \|Distribution of transaction fees.
+            _rpTransactionFeeDistribution <- fromProto =<< cParams ^? ProtoFields.transactionFeeDistribution
+            -- \|Rewards paid from the GAS account.
+            _rpGASRewards <- fromProto =<< cParams ^? ProtoFields.gasRewards
+            return Parameters.RewardParameters{..}
+        -- FIXME: ProtoFields.foundationAccount has type AccountAddress, but the type expects an AccountIndex.
+        _cpFoundationAccount <- Nothing
+        _cpPoolParameters <- do
+            fmap Parameters.PoolParametersV0 . fromProto =<< cParams ^? ProtoFields.minimumThresholdForBaking
+        {- FIXME: The following fields are present in the protobuf message for ChainParametersV0, but there
+                  are no corresponding record fields in the datatype for chain protocol 1.
+
+                    // Minimum threshold for becoming a baker.
+                    Amount minimum_threshold_for_baking = 10;
+                    // Keys allowed to do root updates.
+                    HigherLevelKeys root_keys = 11;
+                    // Keys allowed to do level1 updates;
+                    HigherLevelKeys level1_keys = 12;
+                    // Keys allowed to do parameter updates.
+                    AuthorizationsV0 level2_keys = 13;
```

Comment:
These are not part of chain parameters. It is part of `Updates`,
```haskell
-- |Current state of updatable parameters and update queues.
data Updates' (cpv :: ChainParametersVersion) = Updates
    { -- |Current update authorizations.
      _currentKeyCollection :: !(Hashed (UpdateKeysCollection cpv)),
      -- |Current protocol update.
      _currentProtocolUpdate :: !(Maybe ProtocolUpdate),
      -- |Current chain parameters.
      _currentParameters :: !(ChainParameters' cpv),
      -- |Pending updates.
      _pendingUpdates :: !(PendingUpdates cpv)
    }
    deriving (Show, Eq)
```
Specifically the update keys collection.